service/worker/migration/activities.go

```diff
@@ -252,24 +254,37 @@ func (a *activities) checkHandoverOnce(ctx context.Context, waitRequest waitHand
 	return readyShardCount == len(resp.Shards), nil
 }
 
-func (a *activities) generateWorkflowReplicationTask(ctx context.Context, wKey definition.WorkflowKey) error {
+func (a *activities) generateWorkflowReplicationTask(ctx context.Context, rateLimiter quotas.RateLimiter, wKey definition.WorkflowKey) error {
+	if err := rateLimiter.WaitN(ctx, 1); err != nil {
+		return err
+	}
+
 	// will generate replication task
 	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
 	defer cancel()
 
-	_, err := a.historyClient.GenerateLastHistoryReplicationTasks(ctx, &historyservice.GenerateLastHistoryReplicationTasksRequest{
+	resp, err := a.historyClient.GenerateLastHistoryReplicationTasks(ctx, &historyservice.GenerateLastHistoryReplicationTasksRequest{
 		NamespaceId: wKey.NamespaceID,
 		Execution: &commonpb.WorkflowExecution{
 			WorkflowId: wKey.WorkflowID,
 			RunId:      wKey.RunID,
 		},
 	})
 
-	if _, isNotFound := err.(*serviceerror.NotFound); isNotFound {
-		// ignore NotFound error
+	switch err.(type) {
+	case nil:
+		stateTransitionCount := resp.StateTransitionCount
+		for stateTransitionCount > 0 {
```

Comment:
puzzled by the need of `for` loop here. not knowing how rateLimiter is implemented, would `ReserveN(, stateTransitionCount)` fail if `stateTransitionCount > rateLimiter.Burst()`?

Reply:
yes, the returned reservation will return Ok() == false

Reply:
Do we know the upper bound of ratio between stateTransitionCount and Replication Tasks? If we know, say X, we could probably set burst to X*RPS instead.